src/script/engines/lua.c

```diff
@@ -415,6 +482,14 @@ struct mScriptEngineContext* _luaCreate(struct mScriptEngine2* engine, struct mS
 	lua_getglobal(luaContext->lua, "require");
 	luaContext->require = luaL_ref(luaContext->lua, LUA_REGISTRYINDEX);
 
+#if LUA_VERSION_NUM < 503
+	lua_pushcfunction(luaContext->lua, _luaIpairs);
+	lua_setglobal(luaContext->lua, "ipairs");
+
+	lua_pushcfunction(luaContext->lua, _luaPairs);
+	lua_setglobal(luaContext->lua, "pairs");
```

Comment:
Aren't pairs/ipairs builtins in 5.2?

Reply:
They're builtins in 5.1 and LuaJIT so I'm not sure what this is for.

Reply:
These lines are added in order to replace the original functions with the ones defined in this file. I don't know if there's a better method for this.

Reply:
The only thing that comes to mind is that PUC-Rio had removed ipairs from prerelease versions of 5.2 before re-adding it in response to overwhelming outcry from the community. If this code is derived from some _ancient_ 5.1/5.2 compatibility hack then that might explain it, but pairs was never on the chopping block so I really don't know.